packages/block-library/src/heading/style.scss

```diff
@@ -21,3 +22,27 @@ h6 {
 		padding: $block-bg-padding--v $block-bg-padding--h;
 	}
 }
+
+:root h1 {
+	font-size: calc(1px * var(--wp--typography--font-base) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale));
+}
+
+:root h2 {
+	font-size: calc(1px * var(--wp--typography--font-base) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale));
+}
+
+:root h3 {
+	font-size: calc(1px * var(--wp--typography--font-base) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale));
+}
+
+:root h4 {
+	font-size: calc(1px * var(--wp--typography--font-base) * var(--wp--typography--font-scale) * var(--wp--typography--font-scale));
+}
+
+:root h5 {
+	font-size: calc(1px * var(--wp--typography--font-base) * var(--wp--typography--font-scale));
+}
+
+:root h6 {
+	font-size: calc(0.5px * var(--wp--typography--font-base) * var(--wp--typography--font-scale));
```

Comment:
I think there is some probability that they will break many teams.
We are only allowing to use global styles if theme has some kind of support, but we are unconditionally adding the styles that use the global styles variables.
I think given that styles are unconditional the global styles should also be. So, at least themes have a way to fix things using theme.json if the result is not expected.

Reply:
This PR adds font-size and font-weight to existing blocks that have already added CSS to target line-height and/or colors as CSS variables. Is there any reason we should treat font-size or font-weight differently?

Reply:
For line height, and colors we use the variables but these variables can be defined by the user. E.g: the user can use the UI to change line-height, and colors.

Here we use the variables and don't allow any control over the variables in most themes.
I think we should uncondiotanily show the global styles UI because we unconditionally add the styles that use the variables.

Reply:
Ah, I see what you mean, thanks for the clarification :+1:

Here's my thinking:

- I think we can land this PR without user-facing UI controls: UI controls at the block level are nice, but they don't fix the potential issues. Example: for colors, the vars are only used when the users choose a _custom_ color but the CSS is always present so the theme still needs to account for when the user doesn't modify anything. However! I've already started adapting [the UI controls to work with font-size](https://github.com/WordPress/gutenberg/pull/21153), I just don't think that work should block this.

- As to whether to always load the global styles mechanism. Your point is solid: if we already ship block's CSS with variables, we should allow themes to set them. I may be on board with that, let's discuss it separately at https://github.com/WordPress/gutenberg/pull/21346